app/scripts/web/app.py

```
@@ -0,0 +1,1220 @@
+        soup = _BS(resp.text, "html.parser")
+        # Extract title
+        title_tag = soup.find("title")
+        title = title_tag.get_text(" ", strip=True) if title_tag else url
+        # Remove chrome
+        for tag in soup.find_all(["script", "style", "nav", "header", "footer",
+                                   "aside", "noscript", "button", "form"]):
+            tag.decompose()
+        # Get main text
+        tags = soup.find_all(["h1","h2","h3","h4","p","li","blockquote","td","th","pre","code","dt","dd"])
+        lines = [t.get_text(" ", strip=True) for t in tags]
+        text = "\n".join(l for l in lines if len(l) > 20)
```

Ruff (0.15.5)

[error] Ambiguous variable name: `l`

(E741)